postpone/postpone.c

```diff
@@ -418,13 +417,19 @@ static int create_tmp_files_for_attachments(FILE *fp_body, struct Buffer *file,
       if (!s.fp_out)
         return -1;
 
-      if (((WithCrypto & APPLICATION_PGP) != 0) &&
-          ((sec_type = mutt_is_application_pgp(b)) & (SEC_ENCRYPT | SEC_SIGN)))
+      SecurityFlags sec_type = SEC_NO_FLAGS;
+      if (((WithCrypto & APPLICATION_PGP) != 0) && !sec_type)
+        sec_type = mutt_is_application_pgp(b);
+      if (((WithCrypto & APPLICATION_SMIME) != 0) && !sec_type)
+        sec_type = mutt_is_application_smime(b);
+      if (sec_type & (SEC_ENCRYPT | SEC_SIGN))
```

Comment:
I’m not convinced by the use of `sec_type` as a bool, using the fact that `SEC_NO_FLAGS` is 0. Wouldn’t it be better to test explicitely?